### PR TITLE
Remove invalid Kafka tests [HZ-3675] [HZ-2779]

### LINKED ISSUE #24852: com.hazelcast.jet.kafka.impl.StreamKafkaPTest.when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart [HZ-2779]
_master_ (commit aab1256d2d6c2efbfa45fd270e1fb8019192fd6c)

Failed on Windows - MicrosoftJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Windows-MicrosoftJDK17/68/testReport/com.hazelcast.jet.kafka.impl/StreamKafkaPTest/when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<300> but was:<311>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.lambda$testWithJobRestart$4(StreamKafkaPTest.java:350)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1165)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1184)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.testWithJobRestart(StreamKafkaPTest.java:350)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.testWithJobRestart(StreamKafkaPTest.java:315)
	at com.hazelcast.jet.kafka.impl.StreamKafkaPTest.when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart(StreamKafkaPTest.java:249)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-Windows-MicrosoftJDK17/68/

### LINKED ISSUE #25912: com.hazelcast.jet.kafka.impl.StreamKafkaPTest.when_processingGuaranteeNoneWithConsumerGroup_then_continueFromLastReadMessageAfterJobRestart [HZ-3675]
Branch: [semihbkgr:cn-559-backport-5.3.z](https://github.com/semihbkgr/hazelcast/tree/cn-559-backport-5.3.z) 
Commit: [d7522e26adde4c42e4308427130e124a218dfbbf](https://github.com/semihbkgr/hazelcast/tree/d7522e26adde4c42e4308427130e124a218dfbbf)

PR: https://github.com/hazelcast/hazelcast/pull/25845

https://github.com/hazelcast/hazelcast/pull/25845#issuecomment-1791258042

Standard output can be found here - https://s3.console.aws.amazon.com/s3/object/j-artifacts?region=us-east-1&prefix=Hazelcast-pr-builder/21526/Hazelcast-pr-builder-21526.zip

### PR DESCRIPTION
**Pull request by TomaszGaweda**
For processing guarantee NONE we cannot guarantee any specific output, so removing those tests.

Fixes #25912
Fixes #24852

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [ ] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by TomaszGaweda** [comment]
FYI @anestoruk

File: extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java [hunk 1]
```diff
@@ -169,11 +169,6 @@ public void when_projectionFunctionProvided_then_appliedToReadRecords() {
         });
     }
 
-    @Test
-    public void when_processingGuaranteeNone_then_ignorePartitionsInitialOffsets() {
```
Comment (k-jamroz) [inline extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java:173]: is it possible to test those scenarios with processing guarantee NONE? The scenarios were fine but it cannot be tested by only observing the output of the pipeline. Maybe we should just temporarily ignore the tests until we have better way to test that?

Comment (TomaszGaweda) [reply extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java:173]: I don't think so. We can test if there are any rows processed, but I don't think that's a good test

**Review by k-jamroz — APPROVED** [review]

**Comment by TomaszGaweda** [comment]
run-lab-run

File: extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java (hunk 1 above)
Comment (anestoruk) [reply extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java:173]: Is it necessary to delete this particular test? I think only test scenarios that restart the job are affected by these random failures i.e. :

- `when_processingGuaranteeNone_then_continueFromBeginningAfterJobRestart`
- `when_processingGuaranteeNoneWithConsumerGroup_then_continueFromLastReadMessageAfterJobRestart`

(but I might be wrong)

Comment (TomaszGaweda) [reply extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java:173]: With processingGuarantee NONE you cannot guarantee any behaviour in regard to which items will be reprocessed and which not

**Pull request merged**